Digital-Architects-Avans/rmc-app
# Implemented Edit my account functionality

# User Profile Editing Feature

## Description
This pull request introduces a new feature that allows users to edit their profiles and save the changes to the database. Additionally, it ensures that the correct user name appears on the `MyAccount` screen after a change has been made to the user profile.

## Changes Made

### MyAccount Screen
- Updated the UI to display the latest user details fetched from the database.
- Integrated the new `EditMyAccount` screen for editing user profile functionality.
- Ensured that the UI updates reflect the changes made to the user profile.

### EditMyAccount Screen
- Added logic to fetch the current user details and pre-fill the edit form.
- Implemented user input validation and updating the local state accordingly.
- Integrated the confirmation button logic to update the user profile in the database.
- Introduced a delay to handle UI updates before navigating back to the `MyAccount` screen.


## How to Test
1. Navigate to the `MyAccount` screen.
2. Observe the existing user details.
3. Click on the profile picture to access the `EditMyAccount` screen.
4. Make changes to the user profile fields.
5. Click on the "Apply" button.
6. Observe that the changes are reflected on the `MyAccount` screen with the correct user name.

## Additional Notes
Had to make use of `StateFlow<Boolean>` to add a listener so the navigation would only take place after the UI state has been updated to resolve a bug where you would navigate back to the My Account screen before the UI had been updated resulting in an out of date first name of the user
